dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -515,93 +541,111 @@ class DALIGluonIterator(_DALIMXNetIteratorBase):
+
+    last_batch_policy = PARTIAL, last_batch_padded = False -> last batch = ``[7]``, next iteration will return ``[2, 3]``
 
-    fill_last_batch = False, last_batch_padded = False -> last batch = ``[7]``, next iteration will return ``[2, 3]``
+    last_batch_policy = FILL, last_batch_padded = True   -> last batch = ``[7, 7]``, next iteration will return ``[1, 2]``
 
-    fill_last_batch = True, last_batch_padded = True   -> last batch = ``[7, 7]``, next iteration will return ``[1, 2]``
+    last_batch_policy = FILL, last_batch_padded = False  -> last batch = ``[7, 1]``, next iteration will return ``[2, 3]``
 
-    fill_last_batch = True, last_batch_padded = False  -> last batch = ``[7, 1]``, next iteration will return ``[2, 3]``
+    last_batch_policy = DROP, last_batch_padded = True   -> last batch = ``[5, 6]``, next iteration will return ``[1, 2]``
 
+    last_batch_policy = DROP, last_batch_padded = False  -> last batch = ``[5, 6]``, next iteration will return ``[2, 3]``
     """
     def __init__(self,
                  pipelines,
                  size=-1,
                  reader_name=None,
                  output_types=None,
                  auto_reset=False,
-                 fill_last_batch=True,
-                 last_batch_padded=False):
+                 fill_last_batch=None,
+                 last_batch_padded=False,
+                 last_batch_policy=LastBatchPolicy.FILL):
+
+        # check the assert first as _DaliBaseIterator would run the prefetch
+        self._output_tags = {DALIGluonIterator.DENSE_TAG, DALIGluonIterator.SPARSE_TAG}
+        assert output_types is None or set(output_types) <= self._output_tags, \
+            "Only DENSE_TAG and SPARSE_TAG are allowed"
+
+        self._outputs_types = output_types
+
         super(DALIGluonIterator, self).__init__(
             pipelines,
             size,
             reader_name,
             fill_last_batch,
             last_batch_padded,
-            auto_reset)
+            auto_reset,
+            last_batch_policy)
 
         self._data_batches = [None for i in range(self._num_gpus)]
-        self._output_tags = {DALIGluonIterator.DENSE_TAG, DALIGluonIterator.SPARSE_TAG}
-        assert output_types is None or set(output_types) <= self._output_tags, \
-            "Only DENSE_TAG and SPARSE_TAG are allowed"
-
-        self._outputs_types = output_types
-
-        # We need data about the batches (like shape information),
-        # so we need to run a single batch as part of setup to get that info
-        for p in self._pipes:
-            with p._check_api_type_scope(types.PipelineAPIType.ITERATOR):
-                p.schedule_run()
 
+        self._first_batch = None
+        try:
+            self._first_batch = self.next()
+        except StopIteration:
+            assert False, "It seems that there is not data in the pipeline, please check if last_batch_policy is set PARTIAL and batch size is bigger than the shard size"
```

Comment:
```suggestion
            assert False, "It seems that there is no data in the pipeline. This may happen if `last_batch_policy` is set to PARTIAL and the requested batch size is greater than the shard size."
```

Reply:
Done